clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy

```diff
@@ -0,0 +1,208 @@
+        .build())
+    }
+  }
+
+  @Override
+  LbPoolV2 getPool(final String region, final String id) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().lbPool().get(id)
+    }
+  }
+
+  @Override
+  LbPoolV2 updatePool(final String region, final String id, final String method) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().lbPool().update(id, Builders.lbPoolV2Update()
+        .lbMethod(LbMethod.forValue(method))
+        .adminStateUp(Boolean.TRUE)
+        .build())
+    }
+  }
+
+  @Override
+  ActionResponse deletePool(final String region, final String id) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().lbPool().delete(id)
+    }
+  }
+
+  @Override
+  ActionResponse deleteMonitor(final String region, final String id) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().healthMonitor().delete(id)
+    }
+  }
+
+  @Override
+  HealthMonitorV2 getMonitor(final String region, final String id) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().healthMonitor().get(id)
+    }
+  }
+
+  @Override
+  HealthMonitorV2 createMonitor(final String region, final String poolId, final HealthMonitor healthMonitor) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().healthMonitor().create(Builders.healthmonitorV2()
+        .poolId(poolId)
+        .type(HealthMonitorType.forValue(healthMonitor.type?.name()))
+        .delay(healthMonitor.delay)
+        .expectedCodes(healthMonitor.expectedCodes?.join(','))
+        .httpMethod(healthMonitor.httpMethod)
+        .maxRetries(healthMonitor.maxRetries)
+        .timeout(healthMonitor.timeout)
+        .urlPath(healthMonitor.url)
+        .adminStateUp(Boolean.TRUE)
+        .build())
+    }
+  }
+
+  @Override
+  HealthMonitorV2 updateMonitor(final String region, final String id, final HealthMonitor healthMonitor) {
+    handleRequest {
+      getRegionClient(region).networking().lbaasV2().healthMonitor().update(id, Builders.healthMonitorV2Update()
+        .delay(healthMonitor.delay)
+        .expectedCodes(healthMonitor.expectedCodes?.join(','))
+        .httpMethod(healthMonitor.httpMethod)
+        .maxRetries(healthMonitor.maxRetries)
+        .timeout(healthMonitor.timeout)
+        .urlPath(healthMonitor.url)
+        .adminStateUp(Boolean.TRUE)
+        .build())
+    }
+  }
+
+  @Override
+  Port getPort(final String region, final String portId) {
+    handleRequest {
+      getRegionClient(region).networking().port().get(portId)
+    }
+  }
+
+  @Override
+  Port updatePort(final String region, final String portId, final List<String> securityGroups) {
+    handleRequest {
+      // Builder doesn't take in list of security groups and doesn't allow you to set the ID so, adding some ugly code :)
+      PortBuilder portBuilder = Builders.port()
+      securityGroups.each { portBuilder = portBuilder.securityGroup(it) }
```

Comment:
Are you replacing `portBuilder` each time or is it just adding all the security groups to the port builder?